server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3520,27 +3520,17 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                     }
                     rootDiskSize = Long.parseLong(customParameters.get("rootdisksize"));
 
-                    // only KVM supports rootdisksize override
-                    if (hypervisorType != HypervisorType.KVM) {
```

Comment:
@pdube There was no resource layer code available for XS for this and root volume resize was only implemented for KVM.